test/e2e/storage/csi_mock_volume.go

```diff
@@ -793,8 +793,8 @@ func checkPodLogs(cs clientset.Interface, namespace, driverPodName, driverContai
 	if foundAttributes.Len() != 0 {
 		return fmt.Errorf("some unexpected volume attributes were found: %+v", foundAttributes.List())
 	}
-	if numNodePublishVolume != numNodeUnpublishVolume {
-		return fmt.Errorf("number of NodePublishVolume %d != number of NodeUnpublishVolume %d", numNodePublishVolume, numNodeUnpublishVolume)
+	if numNodePublishVolume > 0 && numNodeUnpublishVolume == 0 {
```

Comment:
Should we also fail if numNodePublish == 0?

Reply:
if all callers expect a non-zero number of publish/unpublish calls, then yes

Reply:
Updated the test to fail separately for numNodePublish and numNodeUnpublish being 0.  places from where this function is being called, there should always be non-zero number of nodepublish and nodeunpublish cals.